pkg/apis/tenancy/v1alpha1/types.go

```diff
@@ -204,23 +204,34 @@ type ClusterWorkspaceTypeSpec struct {
 	//
 	// +optional
 	DefaultChildWorkspaceType ClusterWorkspaceTypeReference `json:"defaultChildWorkspaceType,omitempty"`
+	// allowAnyChildWorkspaceTypes permits all ClusterWorkspaceTypes to be
+	// created in a workspace of this type.
+	//
+	// +optional
+	AllowAnyChildWorkspaceTypes bool `json:"allowAnyChildWorkspaceTypes,omitempty"`
 
 	// allowedChildWorkspaceTypes is a list of ClusterWorkspaceTypes that can be
 	// created in a workspace of this type.
 	//
 	// By default, no type is allowed. This means no other workspace can be nested
-	// within a workspace of the given type. The name `*` allows any child type to
-	// be nested.
+	// within a workspace of the given type. Use allowAnyChildWorkspaceTypes to be
+	// permissive with child types.
 	//
 	// +optional
 	AllowedChildWorkspaceTypes []ClusterWorkspaceTypeReference `json:"allowedChildWorkspaceTypes,omitempty"`
 
+	// allowAnyParentWorkspaceTypes permits this type to be created in workspaces
+	// of all other ClusterWorkspaceTypes.
+	//
+	// +optional
+	AllowAnyParentWorkspaceTypes bool `json:"allowAnyParentWorkspaceTypes,omitempty"`
```

Comment:
my comment still stands: the intention is good, but this API now stutters. Make it
```
allowedParents:
  anyType: true # either this
  types: [...] # or this
```